docs/root/api/http.rst

```diff
@@ -0,0 +1,154 @@
+    .onHeaders { headers, statusCode, _ ->
+      Log.d("MainActivity", "Received status: " + statusCode + " and headers: " + headers)
+      Unit
+    }
+    .onData { buffer, endStream ->
+      if (endStream) {
+        Log.d("MainActivity", "Finished receiving body data")
+      } else {
+        Log.d("MainActivity", "Received body data, more incoming")
+      }
+      Unit
+    }
+    .onError { error ->
+      Log.d("MainActivity", "Error received: " + error.message)
+      Unit
+    }
+    ...
+
+**Swift**::
+
+  let handler = ResponseHandler()
+    .onHeaders { headers, statusCode, _ in
+      print("Received status: \(statusCode) and headers: \(headers)")
+    }
+    .onData { buffer, endStream in
+      if endStream {
+        print("Finished receiving body data")
+      } else {
+        print("Received body data, more incoming")
+      }
+    }
+    .onError { error in
+      print("Error received: \(error.message)")
+    }
+    ...
+
+---------------
+``RetryPolicy``
+---------------
+
+The ``RetryPolicy`` type allows for customizing retry rules that should be applied to an outbound
+request. These rules are added by calling ``addRetryPolicy(...)`` on the ``RequestBuilder``, and
+are applied when the request is sent.
+
+For full documentation of how these retry rules perform, see Envoy's documentation:
+
+- `Automatic retries <https://www.envoyproxy.io/learn/automatic-retries>`_
+- `Retry semantics <https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/http/http_routing.html?highlight=exponential#retry-semantics>`_
+
+-----------------
+``StreamEmitter``
+-----------------
+
+Once a ``Request`` and ``ResponseHandler`` have been created, a stream can be opened using an
+`Envoy instance <_starting_envoy>`_.
+
+Doing so returns a ``StreamEmitter`` which allows the sender to write to the stream,
+close the stream, send trailers, etc.
+
+**Kotlin**::
+
+  val envoy = AndroidEnvoyClientBuilder(...).build()
+
+  val request = RequestBuilder(...).build()
+  val responseHandler = ResponseHandler(...)
+  val emitter = envoy.send(request, responseHandler)
+
+  emitter.sendData(...)
+  emitter.sendData(...)
+  emitter.close(...)
+
+**Swift**::
+
+  let envoy = try EnvoyClientBuilder(...).build()
+
+  let request = RequestBuilder(...).build()
+  let responseHandler = ResponseHandler(...)
+  let emitter = envoy.send(request, handler: responseHandler)
+
+  emitter.sendData(...)
+  emitter.sendData(...)
+  emitter.close(...)
+
+--------------
+Unary Requests
+--------------
+
+As mentioned above, unary requests are made using the same types that perform streaming requests.
+
+Sending a unary request may be done by either closing the ``StreamEmitter`` after the
+set of headers/data has been written, or by using the helper function that returns a
+``CancelableStream`` type instead of a ``StreamEmitter``.
+
+The unary helper function takes optional body data, closes the stream, then waits for a response.
```

Comment:
nit: it is a little off to say waits for a response. Makes me think as the reader that that is a blocking call.

Reply:
I'll remove it